daemon/ZilliqaDaemon.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2019 Zilliqa
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#ifndef ZILLIQA_DAEMON_ZILLIQADAEMON_H_
+#define ZILLIQA_DAEMON_ZILLIQADAEMON_H_
+
+#include <dirent.h>
+#include <errno.h>
+#include <signal.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+#include <algorithm>
+#include <array>
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
+#include <chrono>
+#include <cstdio>
+#include <ctime>
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <thread>
+#include <unordered_map>
+#include <vector>
+
+class ZilliqaDaemon {
+ public:
+  ZilliqaDaemon(int argc, const char* argv[], std::ofstream* log);
+  void MonitorProcess();
+
+ private:
+  std::ofstream* m_log;
```

Comment:
A better alternative is to use reference here
```suggestion
  std::ofstream& m_log;
```
For the initialization, using member initialization list will allow you to set the initial value.